app/src/main/java/com/kms/cura/view/fragment/HealthTrackerFragment.java

```diff
@@ -105,4 +88,50 @@ private void resetAdapter(ArrayList<String> currentHealth, ArrayList<String> pas
         adapter.resetAdapter(currentHealth, pastHealth);
         adapter.notifyDataSetChanged();
     }
+
+    private void showProgressDialog() {
+        pDialog = new ProgressDialog(getActivity());
+        pDialog.setMessage("Loading...");
+        pDialog.setCancelable(false);
+        pDialog.show();
+    }
+
+    private void hideProgressDialog() {
+        pDialog.dismiss();
+    }
+
+    private ArrayList<String> getNameList(List<HealthEntity> list) {
+        ArrayList<String> names = new ArrayList<>();
+        for (int i = 0; i < list.size(); i++) {
+            names.add(list.get(i).getName());
+        }
+        return names;
+    }
+
+    private ArrayList<HealthEntity> getCurrentHealth(List<HealthEntity> list) {
+        ArrayList<HealthEntity> current = new ArrayList<>();
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getEndDate() == null) {
```

Comment:
Should put getEndDate() == null inside a method called isPastHealth inside HealthEntity